Assets/Scripts/Entity/Player/Player.cs

```diff
@@ -13,11 +13,17 @@ public class Player : Entity
     public void SetEnabled(bool enabled) { actionsEnabled = enabled; }
     public bool ActionsEnabled() { return actionsEnabled; }
 
+    public float invincibilityDurationSeconds;
+    private bool invincible = false;
+    private MeshRenderer playerRenderer;
+
 
     /* Called before the game starts. Sets up all necessary info.
      */ 
     void Awake()
     {
+        GameObject model = gameObject.transform.Find("Model").gameObject;
+        playerRenderer = model.transform.Find("Ghost").gameObject.GetComponent<MeshRenderer>();
```

Comment:
Question: will the model name ever change over the course of the game, such as after possession? Or will the object simply be replaced with a different mesh but retain its own name? If the latter, this should be fine.

Reply:
No, the model name should stay as `Model`. At least that should be easy enough to maintain. Though, model can contain anything. This code is specifically hardcoded to search for the Ghost object and alter it's mesh renderer. There are 2 issues with this

1. Enemies are not expected to have a "Ghost" model object.
2. This implies that the entity model consists solely of one game object ("Ghost" in this case). But we may have more complex enemies with multiple model components (Like a sphere, and a capsule).

This approach should be redone, or at the very least, the implications of such an approach should be carefully considered.